include/stdexec/execution.hpp

```diff
@@ -4634,23 +4634,22 @@ namespace stdexec {
           start(__op_state.__state1_);
         }
 
-        void __complete() noexcept try {
+        void __complete() noexcept {
+          STDEXEC_ASSERT(!__data_.valueless_by_exception());
           std::visit(
             [&]<class _Tup>(_Tup& __tupl) -> void {
               if constexpr (same_as<_Tup, std::monostate>) {
                 std::terminate(); // reaching this indicates a bug in schedule_from
               } else {
                 std::apply(
                   [&]<class... _Args>(auto __tag, _Args&... __args) -> void {
-                    __tag((_Receiver&&) __rcvr_, (_Args&&) __args...);
+                    __try_call(
+                      (_Receiver &&) __rcvr_, __tag, (_Receiver &&) __rcvr_, (_Args &&) __args...);
                   },
                   __tupl);
               }
             },
             __data_);
-        } catch (...) {
-
-          set_error((_Receiver&&) __rcvr_, std::current_exception());
```

Comment:
The `__complete` function isn't doing anything that can fail with an exception. In the bad old days, `set_value` could throw, which is probably why this `try`/`catch` is here, but it can simply be removed now. No need for `__try_call`.

Reply:
Yep, I figured the try-catch was a leftover for `set_value` previously being potentially throwing. However, the `__try_call` is still needed for potentially throwing value type copies, right?

> If decay-copying those values are potentially throwing, then set_error_t(std::exeption_ptr) should be added to the list of possible completions.

Or am I misunderstanding this comment:

> No need for __try_call.

?

Reply:
The concepts ensure this can never happen. The `connect` function requires that a receiver is capable of receiving all the completions of the sender. If a `set_value` expression is potentially throwing, then the concept check fails. The sender cannot connect to such a receiver.

See: https://godbolt.org/z/9Pohne6nr

Reply:
Huh, TIL, thanks for explaining! I didn't realize it applied to the whole expression but now that you spell it out it certainly makes sense.